src/main/java/com/marklogic/client/ext/schemasloader/impl/QbvDocumentFileProcessor.java

```diff
@@ -0,0 +1,143 @@
+
+	public static final String QBV_COLLECTION = "http://marklogic.com/xdmp/qbv";
+	public static final String QBV_XML_PLAN_NAMESPACE = "http://marklogic.com/plan";
+	public static final String QBV_XML_ROOT_ELEMENT = "query-based-view";
+	public static final String JAVASCRIPT_EVAL_TEMPLATE = "declareUpdate(); xdmp.invokeFunction(function() {'use strict'; const op = require('/MarkLogic/optic'); return %s }, {database: xdmp.database('%s')})";
+	public static final String XQUERY_EVAL_TEMPLATE = "xquery version \"1.0-ml\"; import module namespace op=\"http://marklogic.com/optic\" at \"/MarkLogic/optic.xqy\"; xdmp:invoke-function(function() {%s},<options xmlns=\"xdmp:eval\"><database>{xdmp:database('%s')}</database></options>)";
+
+	final private DatabaseClient databaseClient;
+	final private String qbvGeneratorDatabaseName;
+	final protected List<DocumentFile> qbvFiles = new ArrayList<>();
+	final private XMLDocumentManager docMgr;
+
+
+	/**
+	 * @param databaseClient - a MarkLogic DatabaseClient object for the Schemas database
+	 */
+	public QbvDocumentFileProcessor(DatabaseClient databaseClient, String qbvGeneratorDatabaseName) {
+		this.databaseClient = databaseClient;
+		this.qbvGeneratorDatabaseName = qbvGeneratorDatabaseName;
+		this.docMgr = databaseClient.newXMLDocumentManager();
+	}
+
+	@Override
+	public DocumentFile processDocumentFile(DocumentFile documentFile) {
+		if (isQbvQuery(documentFile)) {
+			// Defer processing until after all other schemas have been processed.
+			qbvFiles.add(documentFile);
+			return null;
+		} else {
+			return documentFile;
+		}
+	}
+
+	private boolean isQbvQuery(DocumentFile documentFile) {
+		String uri = documentFile.getUri();
+		String extension = documentFile.getFileExtension();
+		if (extension != null) {
+			extension = extension.toLowerCase();
+		}
+		return
+			"sjs".equals(extension)
+				|| "js".equals(extension)
+				|| "xqy".equals(extension)
+				|| "xq".equals(extension)
+				|| (uri != null && uri.startsWith("/qbv"));
+	}
+
+	public void processQbvFiles() {
+		qbvFiles.forEach(this::processQbvFile);
+		qbvFiles.clear();
+	}
+
+	public void processQbvFile(DocumentFile qbvFile) {
+		ServerEvaluationCall call = getServerEvaluationCall(qbvFile);
+		if (call != null) {
+			StringHandle handleString = new StringHandle();
+			try {
+				call.eval(handleString);
+				if (Format.XML.equals(handleString.getFormat())) {
+					Element root = new SAXBuilder().build(new StringReader(handleString.get())).getRootElement();
+					if (QBV_XML_ROOT_ELEMENT.equals(root.getName()) & (root.getNamespace() != null && root.getNamespace().getURI().equals(QBV_XML_PLAN_NAMESPACE))) {
+						qbvFile.getDocumentMetadata().getCollections().add(QBV_COLLECTION);
+						String uri = qbvFile.getUri() + ".xml";
+						DOMHandle handleXML = new DOMHandle(new DOMOutputter().output(root.getDocument()));
+						handleXML.setFormat(Format.XML);
+						docMgr.write(uri, qbvFile.getDocumentMetadata(), handleXML);
+					} else {
```

Comment:
Sorry to be picky on these error messages, but users/Support get very frustrated at me if they can't understand an error message from ml-gradle (or at least Damon always did). Comments on these 3 errors:

1. For the first one, we don't have a cause - we have an unexpected XML response. So do a message of "Query-Based view generation failed for file: %s; received unexpected response from server: %s".
2. For the second one, we also don't have a cause, we have another unexpected response. And it's likely due to the user not generating a view in their script. So do "Query-Based View generation failed for file: %s; ensure your Optic script includes a call to generate a view; received unexpected response from server: %s".
3. The third one is fine because we do have a cause, though for consistency, change "failed; file %s;" to "failed for file: %s;".